keras_nlp/layers/transformer_decoder_test.py

```diff
@@ -57,25 +57,54 @@ def test_valid_call_with_mask(self):
         )
 
     def test_get_config_and_from_config(self):
-        decoder = transformer_decoder.TransformerDecoder(
+        decoder1 = transformer_decoder.TransformerDecoder(
+            intermediate_dim=4,
+            num_heads=2,
+        )
+        decoder2 = transformer_decoder.TransformerDecoder(
             intermediate_dim=4,
             num_heads=2,
+            kernel_initializer=initializers.HeNormal(),
+            bias_initializer=initializers.Constant(value=2)
         )
-        config = decoder.get_config()
-        expected_config_subset = {
+
+        config1 = decoder1.get_config()
+        config2 = decoder2.get_config()
+
+        expected_config_subset1 = {
+            "intermediate_dim": 4,
+            "num_heads": 2,
+            "dropout": 0,
+            "activation": "relu",
+            "layer_norm_epsilon": 1e-05,
+            "kernel_initializer": None,
+            "bias_initializer": None,
+        }
+        expected_config_subset2 = {
             "intermediate_dim": 4,
             "num_heads": 2,
             "dropout": 0,
             "activation": "relu",
             "layer_norm_epsilon": 1e-05,
+            "kernel_initializer": initializers.serialize(initializers.HeNormal()),
+            "bias_initializer": initializers.serialize(initializers.Constant(value=2)),
         }
-        self.assertEqual(config, {**config, **expected_config_subset})
 
-        restored_decoder = transformer_decoder.TransformerDecoder.from_config(
-            config,
+        self.assertEqual(config1, {**config1, **expected_config_subset1})
+        self.assertEqual(config2, {**config2, **expected_config_subset2})
+
+        restored_decoder1 = transformer_decoder.TransformerDecoder.from_config(
+            config1,
+        )
+        restored_decoder2 = transformer_decoder.TransformerDecoder.from_config(
+            config2,
+        )
+
+        self.assertEqual(
+            restored_decoder1.get_config(), {**config1, **expected_config_subset1}
         )
         self.assertEqual(
-            restored_decoder.get_config(), {**config, **expected_config_subset}
+            restored_decoder2.get_config(), {**config2, **expected_config_subset2}
         )
 
     def test_one_training_step_of_transformer_encoder(self):
```

Comment:
Once you make the fix above, you could test that an invalid initialzer name will throw. Should like something like...

```
self.assertRaises(ValueError):
    transformer_decoder.TransformerDecoder(
        intermediate_dim=4,
        num_heads=2,
        kernel_initializer="invalid",
    )
```

Reply:
Do you mean something like this?

```
        self.assertRaises(
            ValueError,
            transformer_encoder.TransformerEncoder(
                intermediate_dim=4,
                num_heads=2,
                kernel_initializer="invalid",
            ),
        )
```
 Since the code you've mentioned gives indentation errors but the one I wrote fails the test

Reply:
Just to confirm this syntax also works right?

```
        self.assertRaises(
            ValueError,
            transformer_encoder.TransformerEncoder, intermediate_dim = 4, num_heads = 2, dropout = 0.5, kernel_initializer = "Invalid"
        )
```